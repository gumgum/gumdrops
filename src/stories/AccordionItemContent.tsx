import React from 'react';
import cx from 'classnames';
import { AccordionItemProps } from './Accordion';

export const AccordionItemContent: React.FC<Omit<AccordionItemProps, 'label'>> = ({
    className,
    color,
    size,
    children,
    ...otherProps
}) => {
    const stopClick: React.MouseEventHandler = e => {
        e.stopPropagation();
    };

    const rootClass = cx('gds-accordion__child-item', className, {
        [`gds-accordion__child-item--${color}`]: color
    });

    const titleClass = cx('gds-accordion__child-item-title', {
        [`gds-accordion__child-item-title--${size}`]: size
    });

    return (
        <li className={rootClass} onClick={stopClick} {...otherProps}>
            <h4 className={titleClass}>{children}</h4>
        </li>
    );
};
