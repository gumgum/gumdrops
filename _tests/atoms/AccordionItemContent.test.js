import renderer from 'react-test-renderer';
import React from 'react';
import AccordionItemContent from '../../components/atoms/AccordionItemContent';

test('Expect <AccordionItemContent> to render properly', () => {
    const props = {
        className: 'foo',
        size: 'xs',
        context: 'danger'
    };
    const tree = renderer
        .create(
            <AccordionItemContent {...props}>
                <div>foo</div>
            </AccordionItemContent>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
