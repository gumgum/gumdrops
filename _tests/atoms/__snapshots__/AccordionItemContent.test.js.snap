// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <AccordionItemContent> to render properly 1`] = `
<li
  className="gds-accordion__child-item foo gds-accordion__child-item--danger"
  onClick={[Function]}
>
  <h4
    className="gds-accordion__child-item-title gds-accordion__child-item-title--xs"
  >
    <div>
      foo
    </div>
  </h4>
</li>
`;
